src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -32,7 +32,9 @@ private void VisitTypeSymbolWithAnnotations(TypeSymbolWithAnnotations type, Abst
                         }
                         break;
                     case false:
-                        if (format.CompilerInternalOptions.IncludesOption(SymbolDisplayCompilerInternalOptions.IncludeNonNullableTypeModifier))
+                        if (!typeSymbol.IsStructType() &&
+                            !typeSymbol.IsEnumType() &&
```

Comment:
@cston Are there other kinds of symbols for which a non-null (`!`) annotation doesn't make sense? Maybe generic (type arguments)?

Reply:
I think this should be a test for `.IsValueType()`.

---
In reply to: [179591924](https://github.com/dotnet/roslyn/pull/25928#discussion_r179591924) [](ancestors = 179591924)